frontend/tests/components/session-timeout.test.tsx

```diff
@@ -0,0 +1,108 @@
+import { createRoutesStub } from 'react-router';
+
+import { act, fireEvent, render, screen } from '@testing-library/react';
+import { useIdleTimer } from 'react-idle-timer';
+import { describe, expect, it, vi } from 'vitest';
+
+import type { SessionTimeoutProps } from '~/components/session-timeout';
+import SessionTimeout from '~/components/session-timeout';
+
+describe('SessionTimeout', () => {
+  const mockActivate = vi.fn();
+  const mockIsPrompted = vi.fn();
+  const mockGetRemainingTime = vi.fn();
+
+  vi.mock('react-idle-timer');
+
+  vi.mocked(useIdleTimer, { partial: true }).mockReturnValue({
+    activate: mockActivate,
+    isPrompted: mockIsPrompted,
+    getRemainingTime: mockGetRemainingTime,
+  });
+
+  const setup = (props: Partial<SessionTimeoutProps> = {}) => {
+    const defaultProps = {
+      promptBeforeIdle: 30000,
+      timeout: 60000,
+      onSessionEnd: vi.fn(),
+      onSessionExtend: vi.fn(),
+    };
+
+    const RoutesStub = createRoutesStub([
+      {
+        Component: () => <SessionTimeout {...defaultProps} {...props} />,
+        path: '/',
+      },
+    ]);
+
+    return render(<RoutesStub />);
+  };
+
+  it('should render the session timeout dialog when prompted', () => {
+    mockIsPrompted.mockReturnValue(true);
+
+    setup();
+
+    expect(screen.getByText('session-timeout.header')).toBeInTheDocument();
+    expect(screen.getByText('session-timeout.end-session')).toBeInTheDocument();
+    expect(screen.getByText('session-timeout.continue-session')).toBeInTheDocument();
```

Comment:
```suggestion
    expect(screen.queryByText('gcweb:session-timeout.header')).not.toBeNull();
    expect(screen.queryByText('gcweb:session-timeout.end-session')).not.toBeNull();
    expect(screen.queryByText('gcweb:session-timeout.continue-session')).not.toBeNull();
```

this avoids having to use an extra assertion library (jest-dom)